avocado/utils/lv_utils.py

```diff
@@ -0,0 +1,517 @@
+                                         loop_device,
+                                         process.run("losetup --all").stdout)
+            if ramdisk_filename is not None:
+                ramdisk_filename = ramdisk_filename.group(1)
+
+            for _ in range(10):
+                time.sleep(0.1)
+                result = process.run("losetup -d %s" % loop_device,
+                                     ignore_status=True, shell=True)
+                if "resource busy" not in result.stderr:
+                    if result.exit_status == 0:
+                        LOGGER.debug("Loop device %s deleted", loop_device)
+                    else:
+                        LOGGER.debug(
+                            "%s -> %s", result.command, result.stderr)
+                    break
+
+    if ramdisk_filename is not None:
+        if os.path.exists(ramdisk_filename):
+            os.unlink(ramdisk_filename)
+            LOGGER.debug("Ramdisk filename %s deleted", ramdisk_filename)
+            vg_ramdisk_dir = os.path.dirname(ramdisk_filename)
+
+    if vg_ramdisk_dir is not None:
+        process.run("umount %s" % vg_ramdisk_dir,
+                    ignore_status=True, shell=True)
+        if result.exit_status == 0:
+            LOGGER.debug("Successfully unmounted tmpfs from %s",
+                         vg_ramdisk_dir)
+        else:
+            LOGGER.debug("%s -> %s", result.command, result.stderr)
+
+        if os.path.exists(vg_ramdisk_dir):
+            try:
+                shutil.rmtree(vg_ramdisk_dir)
+                LOGGER.debug("Ramdisk directory %s deleted", vg_ramdisk_dir)
+            except OSError:
+                pass
+
+
+def vg_check(vg_name):
+    """
+    Check whether provided volume group exists.
+    """
+    cmd = "vgdisplay %s" % vg_name
+    try:
+        process.run(cmd, shell=True)
+        LOGGER.debug("Provided volume group exists: %s", vg_name)
+        return True
+    except process.CmdError, exception:
+        LOGGER.error(exception)
+        return False
+
+
+def vg_list():
+    """
+    List available volume groups.
+    """
+    cmd = "vgs --all"
+    vgroups = {}
+    result = process.run(cmd, shell=True)
+    lines = result.stdout.strip().splitlines()
+    if len(lines) > 1:
+        columns = lines[0].split()
+        lines = lines[1:]
+    else:
+        return vgroups
+
+    for line in lines:
+        details = line.split()
+        details_dict = {}
+        index = 0
+        for column in columns:
+            if re.search("VG", column):
+                vg_name = details[index]
+            else:
+                details_dict[column] = details[index]
+            index += 1
+        vgroups[vg_name] = details_dict
+    return vgroups
+
+
+def vg_create(vg_name, pv_list, force=False):
+    """
+    Create a volume group by using the block special devices
+    """
+
+    if vg_check(vg_name):
+        raise Exception("Volume group '%s' already exist" % vg_name)
```

Comment:
Ideally you should create `class LVException(Exception)` and use it inside, or at least use some inherited exception like `AssertionError` or `RuntimeError`.